packages/e2e-tests/plugins/block-api/index.js

```diff
@@ -14,6 +14,21 @@
 		},
 	} );
 
+	registerBlockType( 'e2e-tests/auto-inserted', {
+		title: 'Auto Inserted',
+		description: 'Auto-inserted test block.',
+		category: 'widgets',
+		edit() {
+			return 'I got auto-inserted!';
+		},
+		save() {
+			return 'I got auto-inserted';
+		},
+		autoInsert: {
```

Comment:
This PR basically implements this as a generic block editor API. It is a decent approach, maybe the best one but I'd like us to ask the question of whether that's exactly what we want. 

Basically, with this approach any blocks with this config, is going to be auto-inserted regardless of the block editor instances it's used in right? Could be a post editor, a site editor, a widgets editor...

I've heard of cases where we want to auto Insert blocks but only for specific post types for instance.

This approach can still be used if the block config is "modified" dynamically but I think it's worth asking whether we want to tie this with block registration. Can we imagine another approach? What are the pros and cons.

Reply:
> Not entirely sure how we can avoid that at the moment, unless the auto-insertion is part of the "initial parsing"

That would be neat to send the updated HTML from the server as it would work out of the box with the block editor. It would require changes in REST API to ensure that the post content, site templates, reusable blocks, and template parts get all tweaks. We will need exactly the same logic on the server anyway to support the requirements for site visitors who never lunch the editor.

Reply:
Sending from the server is a good solution too but personally I was referring to the parsing done in the client. Either way should work I think. But yeah we do need the server side logic for the frontend.

Reply:
I figured out you were referring to the client side. That works, too. We need to investigate whether applying changes on the server instead makes sense. The benefit would be that such an auto-inserting block would also integrate nicely with other REST API powered apps.